plugin/framework/acctests: Add root id attribute requirement information (#2098)
hashicorp/terraform-website: content/plugin/framework/acctests.mdx

# Acceptance Tests

> Testing can be implemented using the [acceptance test framework](/plugin/sdkv2/testing/acceptance-tests) shipped with SDKv2.

Reviewer: ```suggestion
You can implement testing with the [acceptance test framework](/plugin/sdkv2/testing/acceptance-tests) shipped with SDKv2.
```

Author: I swear, one of these days, I will properly write this type of documentation with active voice! My apologies you always need to nudge me in that direction. 😅

Writing and running tests is similar to SDKv2 providers, with the following exceptions:

> - [`TestCase`](/plugin/sdkv2/testing/acceptance-tests/testcase): Specify the provider via [`ProtoV6ProviderFactories`](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource#TestCase.ProtoV6ProviderFactories).

Reviewer: ```suggestion
- [`TestCase`](/plugin/sdkv2/testing/acceptance-tests/testcase): Specify the provider with [`ProtoV6ProviderFactories`](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource#TestCase.ProtoV6ProviderFactories).
```

documentation](/plugin/sdkv2/testing/acceptance-tests/testcase) for more
information on using `resource.TestCase`.

## Implement id Attribute

> In SDKv2, resources and data sources automatically included an implicit, root level `id` attribute. The `id` attribute is no longer implicitly added.

Reviewer 2: > The `id` attribute is no longer implicitly added.

Are you referring to the SDK or the framework in this statement?

Author: Ick -- the framework. Let's clarify that!

```suggestion
In SDKv2, resources and data sources automatically included an implicit, root level `id` attribute. In framework, the `id` attribute is no longer implicitly added.
```

> In SDKv2, resources and data sources automatically included an implicit, root level `id` attribute. In framework, the `id` attribute is no longer implicitly added.

Reviewer: ```suggestion
In SDKv2, resources and data sources automatically included an implicit, root level `id` attribute. In the framework, the `id` attribute is not implicitly added.
```
